### PR TITLE
NFDIV-2736

### PR DESCRIPTION
**Pull request by tomxelliott**
### Change description ###
Allow multiple General Applications to be submitted

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-2736

Screenshot 2022-10-26 at 10 58 11

### CODE REVIEW

**Review by amoriartyMOJ — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorGeneralApplication.java [hunk 1]
```diff
@@ -85,6 +90,18 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
         pages.forEach(page -> page.addTo(pageBuilder));
     }
 
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDetails<CaseData, State> details) {
+
+        log.info("{} about to start callback invoked for Case Id: {}", SOLICITOR_GENERAL_APPLICATION, details.getId());
+        final CaseData data = details.getData();
+
+        data.setGeneralApplication(GeneralApplication.builder().build());
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorGeneralApplication.java:98]: is there a reason we are setting it empty object on start ?

File: src/integrationTest/java/uk/gov/hmcts/divorce/solicitor/SolicitorGeneralApplicationIT.java [hunk 2]
```diff
@@ -130,7 +130,7 @@ public void shouldAddDocumentToDocumentsUploadedWhenGeneralApplicationSubmitted(
                     .binaryUrl("GA.pdf/binary")
                     .build()
             )
-            .documentType(DocumentType.D11)
+            .documentType(DocumentType.GENERAL_APPLICATION)
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/divorce/solicitor/SolicitorGeneralApplicationIT.java:133]: can use static imports

File: src/integrationTest/resources/solicitor-general-application-payment-response.json [hunk 3]
```diff
@@ -47,6 +47,37 @@
...
+    "generalApplications": [{
+      "id": "${json-unit.any-string}",
+      "value": {
+        "generalApplicationDocument": {
+          "documentLink": {
+            "document_url": "http://localhost:4200/assets/d11",
+            "document_filename": "GA.pdf",
+            "document_binary_url": "GA.pdf/binary"
+          },
+          "documentDateAdded": "2022-10-26",
```
Comment (nitinprabhu) [inline src/integrationTest/resources/solicitor-general-application-payment-response.json:59]: will this date be dynamic ?

File: src/integrationTest/resources/solicitor-general-application-response.json [hunk 4]
```diff
@@ -8,8 +8,22 @@
...
+    "generalApplications": [{
+      "id": "${json-unit.any-string}",
+      "value": {
+        "generalApplicationDocument": {
+          "documentLink": {
+            "document_url": "http://localhost:4200/assets/d11",
+            "document_filename": "GA.pdf",
+            "document_binary_url": "GA.pdf/binary"
+          },
+          "documentDateAdded": "2022-10-26",
```
Comment (nitinprabhu) [inline src/integrationTest/resources/solicitor-general-application-response.json:22]: same here

**Review by nitinprabhu — APPROVED** [review]

**Pull request closed without merging**